### PR TITLE
Added merge count files per pool

### PR DESCRIPTION
**Pull request by BertGalle**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-cmgg/exomecnv/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/exomedepth/merge_count/main.nf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    for file in $files; do
+        if [ -f ${meta.id}_${meta.chr}.txt ]; then
+            paste ${meta.id}_${meta.chr}.txt <(awk '{print \$5}' \$file) > temp_auto.txt
+            mv temp_auto.txt ${meta.id}_${meta.chr}.txt
+            else
+            cp \$file ${meta.id}_${meta.chr}.txt
+        fi
+    done
+    """
```
Comment (nvnieuwk) [inline modules/local/exomedepth/merge_count/main.nf:21]:
```suggestion
    output:
    tuple val(meta), path("${prefix}.txt")

    script:
    prefix = task.ext.prefix ?: "${meta.id}_${meta.chr}"
    """
    for file in $files; do
        if [ -f ${prefix}.txt ]; then
            paste ${prefix}.txt <(awk '{print \$5}' \$file) > temp_auto.txt
            mv temp_auto.txt ${prefix}.txt
            else
            cp \$file ${prefix}.txt
        fi
    done
    """
```
A small change to make it possible to configure the filename via the the configuration file. This makes it easy to tweak the process without having to go into the code itself.

File: workflows/exomecnv.nf [hunk 2]
```diff
@@ -74,10 +79,32 @@ workflow EXOMECNV {
...
+
+    grouped_counts_auto = COUNT_AUTO.out.count
+        .map { meta, txt ->
+            def new_meta = [id:meta.pool,chr:"autosomal"]
+            [new_meta, txt]
+    }
+    .groupTuple()
+
+    COUNT_MERGE_AUTO (
+    grouped_counts_auto
```
Comment (nvnieuwk) [inline workflows/exomecnv.nf:91]:
```suggestion
        grouped_counts_auto
```
to keep the layout the same everywhere

File: modules/local/exomedepth/merge_count/main.nf (hunk 1 above)
Comment (nvnieuwk) [reply modules/local/exomedepth/merge_count/main.nf:21]: You can set the prefix in a configuration file like this:
```groovy
process {
    withName: EXOMEDEPTH_COUNT_MERGE {
        ext.prefix = ...
    }
}
```

**Review by nvnieuwk — COMMENTED** [review]
Looking good! Some simple suggestions, but congrats on the first PR :partying_face:

**Review by nvnieuwk — APPROVED** [review]
LGTM! (Looks Good To Me)

**Comment by ToonRosseel** [comment]
Well done!

**Pull request merged**